services/src/main/java/org/exoplatform/officeonline/rest/WOPIResource.java

```diff
@@ -809,9 +819,38 @@ private Response refreshLock(EditorConfig config, String lockId) {
    * @param config the config
    * @return the response
    */
-  private Response delete(EditorConfig config) {
-    LOG.warn("WOPI DELETE is not allowed for Office Online for web");
-    return Response.ok().build();
+  private Response delete(EditorConfig config, String lockId) {
+    try {
+      wopiService.delete(config, lockId);
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Delete response: OK");
+      }
+      return Response.ok().build();
+    } catch (LockMismatchException e) {
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("Lock mismatch for delete. Given lock: " + lockId + ", Actual: " + e.getLockId());
+      }
+      return Response.status(Status.CONFLICT)
+                     .entity("{\"error\": \"" + e.getMessage() + "\"}")
+                     .header(LOCK, e.getLockId())
+                     .type(MediaType.APPLICATION_JSON)
+                     .build();
+    } catch (FileNotFoundException e) {
+      if (LOG.isDebugEnabled()) {
+        LOG.debug("File not found for refresh lock", e);
+      }
+      return Response.status(Status.NOT_FOUND)
+                     .entity("{\"error\": \"" + e.getMessage() + "\"}")
+                     .type(MediaType.APPLICATION_JSON)
+                     .build();
+
+    } catch (OfficeOnlineException | RepositoryException e) {
+      LOG.error("Cannot delete file.", e);
```

Comment:
Here in the log message we would say more to let figure out an actual file and user, may be other details - add them (but not to the response).

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/13/commits/e93ddc3e549cd35e3bb596f7c5c6aeb0fdc73878